### PR TITLE
Denys/fix/chat register

### PR DESCRIPTION
**Pull request by DenisBurimov**

### CODE REVIEW

File: app/templates/chat/chat_02_email.html [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<div>
+    {% for message in room.messages %}
+    {% if message.sender_id == 2 %}
```
Comment (Nazar-kob) [inline app/templates/chat/chat_02_email.html:3]: always sender_id == 2

File: app/views/chat.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+def sell():
+    now = datetime.now()
+    now_str = now.strftime("%Y-%m-%d %H:%M")
+
+    question = "Are you looking for buying or selling tickets or for events information?"
+
+    seller_id = current_user.id if current_user.is_authenticated else None
+    room = m.Room(
+        seller_id=seller_id,
+        buyer_id=2,
```
Comment (Nazar-kob) [inline app/views/chat.py:25]: why hard code ? buyer_id=2

File: app/views/chat.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    ).save()
+    m.Message(
+        sender_id=2,
+        room_id=room.id,
```
Comment (Nazar-kob) [inline app/views/chat.py:29]: sender_id=2,

File: app/views/chat.py [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    m.Message(
+        room_id=room.id,
+        text=user_name,
+    ).save(False)
+    user = m.User(
+        username=user_name,
+        email="empty@email.com",
+        phone="00000000000",
+        card="0000000000000000",
+        password="",
```
Comment (Nazar-kob) [inline app/views/chat.py:92]: why hard code?

File: app/views/chat.py [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+
+    room_unique_id = request.args.get("room_unique_id")
+    email = request.args.get("chat_email")
+    user_unique_id = request.args.get("user_unique_id")
+
+    user_query = m.User.select().where(m.User.unique_id == user_unique_id)
+    user: m.User = db.session.scalar(user_query)
+
+    room_query = m.Room.select().where(m.Room.unique_id == room_unique_id)
+    room: m.Room = db.session.scalar(room_query)
```
Comment (Nazar-kob) [inline app/views/chat.py:121]: move the code below if

File: app/views/chat.py [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+
+    room_unique_id = request.args.get("room_unique_id")
+    phone = request.args.get("chat_phone")
+    user_unique_id = request.args.get("user_unique_id")
+
+    user_query = m.User.select().where(m.User.unique_id == user_unique_id)
+    user: m.User = db.session.scalar(user_query)
+
+    room_query = m.Room.select().where(m.Room.unique_id == room_unique_id)
+    room: m.Room = db.session.scalar(room_query)
```
Comment (Nazar-kob) [inline app/views/chat.py:209]: move it below if

File: src/base.ts [hunk 7]
```diff
@@ -88,3 +88,124 @@ scrollTopButton.addEventListener('click', e => {
     behavior: 'smooth',
   });
 });
+
+// register verification modal
+const verificationInputs = document.querySelectorAll(
```
Comment (Nazar-kob) [inline src/base.ts:93]: add code typing

File: test_flask/test_users_chat.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    login(client)
+    response = client.get("/")
+    assert f"Hi, {current_user.username}" in response.data.decode()
+    assert "Are you looking for buying or selling tickets or for events information" in response.data.decode()
+
+
+def test_chat_sell(client: FlaskClient):
+    response = client.get("/chat/sell")
+    assert response.status_code == 200
+    assert len(m.Room.all()) == 1
```
Comment (Nazar-kob) [inline test_flask/test_users_chat.py:17]: please do not hard code

**Review by Nazar-kob — APPROVED** [review]

File: .gitignore [hunk 9]
```diff
@@ -10,3 +10,4 @@ node_modules/
 *.pyc
 /*.sqlite3
 .DS_Store
+ticket-auth-8263c-firebase-adminsdk-x3dnf-2a2ff09f0a.json
```
[the remark this answers is not in the document]
Comment (DenisBurimov) [reply .gitignore:13]: Redundant file. Forgot to remove. Thx

File: app/controllers/jinja_globals.py [hunk 10]
```diff
@@ -20,9 +22,22 @@ def time_delta(created_at: datetime):
     return (now - created_at).days * -1
 
 
-def cut_seconds(created_at: datetime):
+def cut_seconds(created_at: datetime = datetime.now()):
     return created_at.strftime("%Y-%m-%d %H:%M")
 
 
-def card_mask(card_number: str):
+def card_mask(card_number: str = "000000000000000000"):
```
[the remark this answers is not in the document]
Comment (DenisBurimov) [reply app/controllers/jinja_globals.py:29]: Unfortunately we can't do that cause the customer demands do that as in figma

File: app/controllers/jinja_globals.py [hunk 11]
```diff
@@ -20,9 +22,22 @@ def time_delta(created_at: datetime):
...
     return f"{card_number[:4]} **** **** {card_number[-4:]}"
+
+
+def get_categories():
+    return m.Category.all()
```
[the remark this answers is not in the document]
Comment (DenisBurimov) [reply app/controllers/jinja_globals.py:34]: Done. Thank you

File: app/forms/chat.py [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+from app import models as m
+from app import db
+
+
+class ChatPhoneForm(FlaskForm):
+    phone = StringField("phone", [DataRequired()])
+    submit = SubmitField("Save")
+
+    def validate_phone(self, field):
+        query = m.User.select().where(m.User.phone == field.data).where(m.User.id != current_user.id)
```
[the remark this answers is not in the document]
Comment (DenisBurimov) [reply app/forms/chat.py:19]: Done. Thank you

File: app/models/user.py [hunk 13]
```diff
@@ -95,9 +94,7 @@ class User(db.Model, UserMixin, ModelMixin):
         back_populates="user",
     )
 
-    notifications_config: orm.Mapped["NotificationsConfig"] = orm.relationship(
-        back_populates="user"
-    )
+    notifications_config: orm.Mapped["NotificationsConfig"] = orm.relationship(back_populates="user")
```
[the remark this answers is not in the document]
Comment (DenisBurimov) [reply app/models/user.py:97]: Yes

File: app/templates/auth/login.html [hunk 14]
```diff
@@ -1,49 +1,48 @@
...
+  <div class="pb-5 px-6 border-b border-light-grey-b0">
+    <h2 class="text-3xl">Login</h2>
+  </div>
+  <div class="px-6 flex flex-col">
+    <form
+      role="form"
+      action="{{ url_for('auth.login') }}"
+      method="post"
+      class="py-9 flex flex-col gap-4">
+      <input
```
[the remark this answers is not in the document]
Comment (DenisBurimov) [reply app/templates/auth/login.html:17]: Done. Thank you

File: app/templates/auth/phone.html [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+      role="form"
+      action="{{ url_for('auth.activate', reset_password_uid=reset_password_uid) }}"
+      method="post"
+      class="py-9 flex flex-col gap-4">
+      <input
+        name="phone"
+        class="py-4 px-4 rounded-2xl button-gradient-grey"
+        type="text"
+        placeholder="Digite seu número de telefonei" />
+      <button
```
[the remark this answers is not in the document]
Comment (DenisBurimov) [reply app/templates/auth/phone.html:22]: Done. Thank you

File: app/templates/auth/phone_verification.html [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+    <form role="form" action="#" method="post" class="py-9 flex flex-col">
+      <p class="mb-4">Código de verificação</p>
+      <div class="mb-2 flex gap-3">
+        <input name="digit_1"
+          class="auth-register-verification-input flex-1 min-w-0 py-4 px-4 rounded-2xl outline-none button-gradient-grey focus:border-transparent focus:ring-0"
+          type="text" placeholder="" />
+        <input name="digit_2"
+          class="auth-register-verification-input flex-1 min-w-0 py-4 px-4 rounded-2xl outline-none button-gradient-grey focus:border-transparent focus:ring-0"
+          type="text" placeholder="" disabled />
+        <input name="digit_3"
```
[the remark this answers is not in the document]
Comment (DenisBurimov) [reply app/templates/auth/phone_verification.html:20]: Done. Thank you

File: app/templates/auth/register.html [hunk 17]
```diff
@@ -1,78 +1,52 @@
...
+      role="form"
+      action="{{ url_for('auth.register') }}"
+      method="post"
+      class="py-9 px-6 flex flex-col gap-4">
+      <input
+        name="username"
+        class="py-4 px-4 rounded-2xl button-gradient-grey"
+        type="text"
+        placeholder="Nome de usuário" />
+      <input
```
[the remark this answers is not in the document]
Comment (DenisBurimov) [reply app/templates/auth/register.html:20]: Done. Thank you

File: app/views/chat.py (hunk 5 above)
Comment (DenisBurimov) [reply app/views/chat.py:121]: Done. Thank you

File: app/templates/chat/chat_02_email.html (hunk 1 above)
Comment (DenisBurimov) [reply app/templates/chat/chat_02_email.html:3]: Done. Thx

File: app/forms/auth.py [hunk 18]
```diff
@@ -38,6 +38,25 @@ def validate_email(form, field):
...
+            raise ValidationError("This phone number is taken.")
+
+
+class VerificationCodeForm(FlaskForm):
+    digit_1 = IntegerField("Digit", validators=[DataRequired()])
+    digit_2 = IntegerField("Digit", validators=[DataRequired()])
+    digit_3 = IntegerField("Digit", validators=[DataRequired()])
+    digit_4 = IntegerField("Digit", validators=[DataRequired()])
+    digit_5 = IntegerField("Digit", validators=[DataRequired()])
+    digit_6 = IntegerField("Digit", validators=[DataRequired()])
```
[the remark this answers is not in the document]
Comment (DenisBurimov) [reply app/forms/auth.py]: No. We have certain design

**Pull request merged**